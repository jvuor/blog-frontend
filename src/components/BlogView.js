import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { Icon, Segment, Header } from 'semantic-ui-react'
import Moment from 'moment'
import ReactMarkdown from 'react-markdown'

class BlogView extends React.Component {
  componentDidMount () {
    document.title = `${this.props.blogs.find(blog => blog.id === this.props.blogId).title} - Jussin Blogi`
  }

  componentWillUnmount () {
    document.title = 'Jussin Blogi'
  }

  render () {
    const blog = this.props.blogs.find(m => m.id === this.props.blogId)
    return (
      <div>
        <Link to='/'>
          <h6><Icon name='angle double left' />Back</h6>
        </Link>
        <Header icon textAlign='center'>
          <Icon name='envelope open outline' circular /><br />
          <Header.Content>
            {blog.title}
          </Header.Content>
        </Header>
        <ReactMarkdown source={blog.content} />
        <Segment compact>
          <Icon name='pencil' />
          Written by {blog.user.name} {Moment(blog.created).fromNow()}
        </Segment>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    blogs: state.blogs
  }
}

export default connect(mapStateToProps, null)(BlogView)