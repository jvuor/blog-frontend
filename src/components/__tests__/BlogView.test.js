import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { BlogView } from '../BlogView'
import { shallow } from 'enzyme'
import blogTestData from './blogTestData'

describe('BlogList component unit tests', () => {
  test('breaks on an empty redux state', () => {
    const blogs = null
    const blogId = null
    const result = shallow(
      <MemoryRouter>
        <BlogView blogs={blogs} blogId={blogId} />
      </MemoryRouter>
    )

    expect(() => result.html()).toThrow(TypeError)
  })

  test('shows expected data when given a proper state', () => {
    const blogs = blogTestData
    const blogId = '5b58dce799f2274cc85b0933'
    const wrapper = shallow(
      <MemoryRouter>
        <BlogView blogs={blogs} blogId={blogId} />
      </MemoryRouter>
    )

    const content = wrapper.html()

    expect(content).toContain('Markdown Test')
    expect(content).toContain('Test User')
    //check for no markup
    expect(content).not.toContain('#')
    expect(content).not.toContain('*')
  })

  test('snapshot test', () => {
    const blogs = blogTestData
    const blogId = '5b58dce799f2274cc85b0933'
    const wrapper = shallow(
      <MemoryRouter>
        <BlogView blogs={blogs} blogId={blogId} />
      </MemoryRouter>
    )
    const content = wrapper.html()

    expect(content).toMatchSnapshot()
  })
})