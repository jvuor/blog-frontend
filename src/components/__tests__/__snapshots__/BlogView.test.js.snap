// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList component unit tests snapshot test 1`] = `"<div><a href=\\"/\\"><h6><i aria-hidden=\\"true\\" class=\\"angle double left icon\\"></i>Back</h6></a><div class=\\"ui icon center aligned header\\"><i aria-hidden=\\"true\\" class=\\"envelope open outline circular icon\\"></i><br/><div class=\\"content\\">Markdown Test</div></div><div><h3>Subheader</h3><p>Hi, this text has <em>markdown</em>! <strong>Hell yeah</strong></p><p><a href=\\"\\">link</a></p><ul><li>List Item</li><li>Another Item</li></ul><h2>Another Subheader</h2><p><strong>Still marking down like a boss yeaaaaaahhhh</strong></p></div><div class=\\"ui compact segment\\"><i aria-hidden=\\"true\\" class=\\"pencil icon\\"></i>Written by Test User 3 days ago</div></div>"`;
